actions/image-build/build.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/sh
+
+# Wraps the process for configuring, building, and testing an NWX repo
+#
+# Usage:
+#   build.sh
+#
+# Other variables:
+#   cmake_version: the version of cmake being used in the format x.y.z
+#
+
+set -e # Exit with error if any command fails
+
+arch=Linux-x86_64
+cmake_root=$(pwd)/cmake-"${cmake_version}"-"${arch}"
+cmake_command=cmake
+toolchain_file=$(pwd)/toolchain.cmake
+
+
+echo "set(BUILD_TESTING ON)" > "${toolchain_file}"
+{
+  echo "set(CMAKE_CXX_STANDARD 17)"
+  echo 'set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)'
+  echo "set(BUILD_SHARED_LIBS ON)"
+  echo "set(CATCH_ENABLE_COVERAGE ON)"
+  echo "set(CMAKE_PREFIX_PATH $(pwd)/install)"
+  echo 'set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --coverage -std=c++17")'
+  echo 'set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DOMPI_SKIP_MPICXX")'
+  echo 'set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} --coverage")'
+  echo 'set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fprofile-arcs")'
+  echo 'set(CMAIZE_GITHUB_TOKEN '${CMAIZE_GITHUB_TOKEN}')'
```

Comment:
I am not sure if it is a serious or valid concern, but maybe it is better to  avoid writing the `CMAIZE_GITHUB_TOKEN` to a file. We could pass it as an option to cmake directly.

Reply:
> I am not sure if it is a serious or valid concern, but maybe it is better to avoid writing the `CMAIZE_GITHUB_TOKEN` to a file. We could pass it as an option to cmake directly.

You mean we run the cmake command as something like:
`cmake CMAIZE_GITHUB_TOKEN='${CMAIZE_GITHUB_TOKEN}'`?

Reply:
I think the following would work: `cmake -D CMAIZE_GITHUB_TOKEN=${CMAIZE_GITHUB_TOKEN}` But, as I said I am not sure if this is important, or not.

Reply:
> I think the following would work: `cmake -D CMAIZE_GITHUB_TOKEN=${CMAIZE_GITHUB_TOKEN}` But, as I said I am not sure if this is important, or not.

I agree that it's not recommended to write any token into a file. I will make the change in my next PR. However, as here everything is inside a container and will be gone when the job is done, so maybe it is fine. Thanks for your suggestion!

Reply:
IIRC,`-DCMAIZE_GITHUB_TOKEN` simply pre-populates the cache with `CMAIZE_GITHUB_TOKEN`. Since CMake's cache is a non-encrypted text file, I don't think this actually solves the problem. For now, I agree with @yzhang-23 's sentiment in that since things are thrown away afterwards it's not terrible...